packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2784,7 +2784,12 @@ class Navigator extends StatefulWidget {
   /// To pop until a route with a certain name, use the [RoutePredicate]
   /// returned from [ModalRoute.withName].
   ///
-  /// The routes are closed with null as their `return` value.
+  /// If `result` is not specified, then all routes are closed
+  /// with null as their `return` value.
+  /// If `result` is specified, then the last popped route will be closed with
+  /// `result` as its return value.
+  ///
+  /// The `T` type argument is the type of the return value of the last popped route.
```

Comment:
What value is the generic type argument adding? Can we drop this and make the argument `Object? result`?

Reply:
What does the type argument accomplish? It doesn't appear to have any impact beyond restricting the argument.

What is the caller-visible behavior difference between calling `pop<String>(context, 'foo');` and `pop<Object>(context, 'foo');`